test/calico/calico_test.go

```diff
@@ -18,37 +18,60 @@
 package calico
 
 import (
-	"context"
+	"encoding/json"
 	"testing"
 
-	"github.com/projectcalico/libcalico-go/lib/apiconfig"
-	client "github.com/projectcalico/libcalico-go/lib/clientv3"
-	"github.com/projectcalico/libcalico-go/lib/options"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructuredscheme"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
 
 	"github.com/kinvolk/lokomotive/test/components/util"
 )
 
 func TestHostEndpoints(t *testing.T) {
-	// Build Calico client.
-	cac := apiconfig.NewCalicoAPIConfig()
-	cac.Spec.DatastoreType = apiconfig.Kubernetes
-	cac.Spec.Kubeconfig = util.KubeconfigPath(t)
+	// Build rest client so we can do equivalent of 'kubectl get --raw'.
+	config, err := clientcmd.BuildConfigFromFlags("", util.KubeconfigPath(t))
+	if err != nil {
+		t.Fatalf("failed building rest client config: %v", err)
+	}
 
-	c, err := client.New(*cac)
+	config.GroupVersion = &schema.GroupVersion{}
+	config.NegotiatedSerializer = unstructuredscheme.NewUnstructuredNegotiatedSerializer()
+
+	client, err := rest.RESTClientFor(config)
 	if err != nil {
-		t.Fatalf("failed creating Calico client: %v", err)
+		t.Fatalf("failed building rest client: %v", err)
 	}
 
-	// Build list of nodes which has associated HostEndpoint object.
-	hostEndpointList, err := c.HostEndpoints().List(context.TODO(), options.ListOptions{})
+	// This is minimal version of Calico HostEndpoint CRD object which we need to deserialize
```

Comment:
```suggestion
	// This is minimal version of the Calico HostEndpoint CRD object which we need to deserialize
```

Reply:
Can we add some link to the upstream structure here?

Reply:
I'm not sure what I should link to. To https://github.com/projectcalico/libcalico-go/blob/master/lib/apis/crd.projectcalico.org/v1/hostendpoint_types.go#L27? + to K8s resource list? Maybe I just include shell command + returned json sample?

Reply:
I'd link to https://github.com/projectcalico/libcalico-go/blob/release-v3.14/lib/apis/v3/hostendpoint.go#L31

Reply:
or maybe without the line number since it's a branch...

Reply:
Done.